packages/p2p/src/pages/my-profile/my-profile.tsx

```diff
@@ -34,7 +34,12 @@ const MyProfile = () => {
     }, []);
 
     React.useEffect(() => {
-        if (is_poi_poa_verified && !general_store.is_advertiser && !general_store.is_loading) {
+        if (
+            is_poi_poa_verified &&
+            !general_store.is_advertiser &&
+            !general_store.is_loading &&
+            general_store.is_p2p_user === false
+        ) {
```

Comment:
is_p2p_user can either be null (default), false or true. In this case we need to check for false as the user when authenticated but they have no nickname, this value should be false.

If they need to do POI/POA, it will also be false, but the is_poi_poa_verified check will prevent the modal from showing.

Also, because of how we set the is_loading in different places to true, and to false, it will show the nickname modal because at some point, it will become true